pkg/controller/kogitodataindex/resource/kafka.go

```diff
@@ -14,19 +14,160 @@
 
 package resource
 
-import "github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+import (
+	"fmt"
 
-func fromKafkaToStringMap(kafka v1alpha1.KafkaConnectionProperties) map[string]string {
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	kafkabetav1 "github.com/kiegroup/kogito-cloud-operator/pkg/apis/kafka/v1beta1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	kafkaClusterLabel string = "strimzi.io/cluster"
+
+	kafkaURINotFoundError string = "there's no Kafka instance URI found in the namespace and Kafka external URI is not specified, cannot deploy Data Index service"
+	kafkaNotExist         string = "there is no Kafka instance found in the namespace, cannot deploy Kafka Topics"
+
+	kafkaTopicConfigRetentionKey   string = "retention.ms"
+	kafkaTopicConfigSegmentKey     string = "segment.bytes"
+	kafkaTopicConfigRetentionValue string = "604800000"
```

Comment:
They are the default values for topic's creation. Since we don't handle this to users to set upon creation, we're using defaults.

Users could update/edit this later via Strimzi.